test/param_build_test.c

```diff
@@ -126,8 +128,14 @@ static int template_private_test(void)
     BIGNUM *bn = NULL, *bn_res = NULL;
     int res = 0;
 
-    if (!TEST_ptr(bld)
-        || !TEST_true(OSSL_PARAM_BLD_push_uint(bld, "i", 6))
+    if (!TEST_ptr(data1 = OPENSSL_secure_malloc(data1_size))
+            || !TEST_ptr(bld = OSSL_PARAM_BLD_new()))
+        goto err;
+
+    for (j = 0; j < data1_num; j++)
+        data1[j] = -16 * j;
+
+    if (!TEST_true(OSSL_PARAM_BLD_push_uint(bld, "i", 6))
```

Comment:
Should there be some CRYPTO_secure_allocated() calls to test that the relevant params `i` and `bignumber` are securely allocated?

Reply:
The native types are function arguments passed by value, so they can't possibly be in secure memory.

The BNs are checked via the `BN_FLG_SECURE` flag.  The BN structure shouldn't ever be in secure memory but the number bytes themselves can be.

Pauli

Reply:
BN's already supported secure allocation in param_bld.

Reply:
I am sorry for a confusion - I meant `oct_s` instead of `i`.

Reply:
Also there is no check that the bn_res has BN_FLG_SECURE or am I missing something?

Reply:
The secure memory allocation code is checked by `test_secmem`, is it necessary to verify it here too?

Reply:
That tests the low level secure memory allocation. What I am asking here for is to test that the OSSL_PARAM_BLD honors the secure memory allocation of the origin.

Reply:
Okay, I understand finally.  I will add.

Reply:
Done.